src/main/java/org/aksw/word2vecrestful/subset/DataSubsetGenerator.java

```diff
@@ -0,0 +1,122 @@
+package org.aksw.word2vecrestful.subset;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.aksw.word2vecrestful.utils.Cfg;
+import org.aksw.word2vecrestful.word2vec.Word2VecFactory;
+import org.aksw.word2vecrestful.word2vec.Word2VecModel;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+/**
+ * Class to help generate and persist the subsets of a word2vec model
+ * 
+ * @author Nikit
+ *
+ */
+public class DataSubsetGenerator {

```

Comment:
Please document the expected file structure somewhere.

Reply:
Already much better. However, could you please
* add an explanation of the single fields (it is not that much clear what "sd" means and we might even forget where the centroid came from...)
* Use a proper formatting because at the moment, javadoc sees the JSON block as text and won't care about the linebreaks, etc. that make it readable.